src/logging.ts

```diff
@@ -99,21 +132,55 @@ export class Logger {
     const formattedMessage = `${prefix} ${message} ${argString}\n`;
 
     return new Promise<void>((resolve, reject) => {
-      const logWriteStream = createWriteStream(LOGFILE_PATH, { flags: "a" });
-      logWriteStream.once("error", (error) => {
-        console.error("Error writing to log file:", error);
-        logWriteStream.end();
-        reject(error);
-      });
-      logWriteStream.write(Buffer.from(formattedMessage), (error) => {
-        if (error) {
-          console.error("Error writing to log file:", error);
-          logWriteStream.end();
-          reject(error);
-        }
-        // waits for any remaining data to be written to the file before closing the stream
-        logWriteStream.end(() => resolve());
-      });
+      try {
+        const stream = getLogFileStream();
+        stream.write(formattedMessage, (error) => {
+          if (error) {
+            console.error("Error writing to log file:", error);
+            reject(error);
+          } else {
+            resolve();
+          }
+        });
+      } catch (error) {
+        console.error("Unexpected error during log file operations:", error);
+        resolve();
+      }
     });
   }
 }
+
+/** Single stream to allow rotating-file-stream to keep track of file sizes and rotation timing. */
+let logFileStream: RotatingFileStream | undefined;
+
+export function getLogFileStream(): RotatingFileStream {
+  if (!logFileStream) {
+    logFileStream = createStream(rotatingFilenameGenerator, {
+      size: MAX_LOG_FILE_SIZE,
+      maxSize: MAX_LOG_FILE_SIZE,
+      maxFiles: MAX_LOG_FILES,
+      interval: LOG_ROTATION_INTERVAL,
+      path: LOGFILE_DIR,
+      encoding: "utf-8",
+    });
+  }
+  return logFileStream;
+}
+
+function pad(num: number) {
+  return (num > 9 ? "" : "0") + num;
+}
+
+const rotatingFilenameGenerator: Generator = (time: number | Date, index?: number): string => {
+  if (!time) return LOGFILE_NAME;
+
+  if (!(time instanceof Date)) {
+    time = new Date(time);
+  }
+  const month = time.getFullYear() + "" + pad(time.getMonth() + 1);
+  const day = pad(time.getDate());
+  const hour = pad(time.getHours());
+  const minute = pad(time.getMinutes());
+
+  return `${month}/${month}${day}-${hour}${minute}-${index}-${LOGFILE_NAME}`;
```

Comment:
Prob cleaner for world citizens to make in ISO YYYY-MM-DD format. And good lord don't have '/' in a filename. Whoa back down there, Satan.